### ISSUE TITLE
Getting error whenever I open ZeroNet.app on OSX

### ISSUE THREAD

**Issue by jamesalexanderdickerson** [HelloZeroNet/ZeroNet#336]
Server error
Err: Exception: File not allowed: data/1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D/index.html in UiServer.py line 81 > UiRequest.py line 82 > UiRequest.py line 211 > UiRequest.py line 285 > SiteStorage.py line 230 > SiteStorage.py line 250
Please report it if you think this an error.
Details:

{
    "GATEWAY_INTERFACE": "CGI/1.1", 
    "HTTP_ACCEPT": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8", 
    "HTTP_ACCEPT_ENCODING": "gzip, deflate", 
    "HTTP_ACCEPT_LANGUAGE": "en-US,en;q=0.5", 
    "HTTP_CONNECTION": "keep-alive", 
    "HTTP_HOST": "127.0.0.1:43110", 
    "HTTP_USER_AGENT": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:44.0) Gecko/20100101 Firefox/44.0", 
    "PATH_INFO": "/1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D", 
    "QUERY_STRING": "", 
    "REMOTE_ADDR": "127.0.0.1", 
    "REMOTE_PORT": "57155", 
    "REQUEST_METHOD": "GET", 
    "SCRIPT_NAME": "", 
    "SERVER_NAME": "1.0.0.127.in-addr.arpa", 
    "SERVER_PORT": "43110", 
    "SERVER_PROTOCOL": "HTTP/1.1", 
    "SERVER_SOFTWARE": "gevent/1.0 Python/2.7", 
    "arguments": {
        "action": "main", 
        "batch": false, 
        "coffeescript_compiler": null, 
        "config_file": "zeronet.conf", 
        "connected_limit": 15, 
        "data_dir": "data", 
        "debug": false, 
        "debug_socket": false, 
        "disable_encryption": false, 
        "disable_sslcompression": true, 
        "disable_udp": false, 
        "fileserver_ip": "*", 
        "fileserver_port": 15441, 
        "homepage": "1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D", 
        "ip_external": null, 
        "keep_ssl_cert": false, 
        "log_dir": "log", 
        "max_files_opened": 1024, 
        "msgpack_purepython": true, 
        "open_browser": "default_browser", 
        "proxy": null, 
        "size_limit": 10, 
        "stream_downloads": false, 
        "tor": "enable", 
        "tor_controller": "127.0.0.1:9051", 
        "tor_proxy": "127.0.0.1:9050", 
        "trackers": [
            "zero://boot3rdez4rzn36x.onion:15441", 
            "zero://boot.zeronet.io#f36ca555bee6ba216b14d10f38c16f7769ff064e0e37d887603548cc2e64191d:15441", 
            "udp://tracker.coppersurfer.tk:6969", 
            "udp://tracker.leechers-paradise.org:6969", 
            "udp://9.rarbg.com:2710", 
            "http://tracker.aletorrenty.pl:2710/announce", 
            "http://explodie.org:6969/announce", 
            "http://torrent.gresille.org/announce"
        ], 
        "trackers_file": false, 
        "ui_ip": "127.0.0.1", 
        "ui_port": 43110, 
        "ui_restrict": false, 
        "use_openssl": true, 
        "use_tempfiles": false, 
        "verbose": false
    }, 
    "plugins": [
        "AnnounceZero", 
        "CryptMessage", 
        "Newsfeed", 
        "Sidebar", 
        "Stats", 
        "Trayicon", 
        "Zeroname"
    ], 
    "version_gevent": "1.0.2", 
    "version_python": "2.7.11 |Anaconda 2.5.0 (x86_64)| (default, Dec  6 2015, 18:57:58) \n[GCC 4.2.1 (Apple Inc. build 5577)]", 
    "version_zeronet": "0.3.6 r948", 
    "wsgi.url_scheme": "http"
}

**Comment by Erkan-Yilmaz** [comment]
What happens when you just go to:
http://127.0.0.1:43110 
or
http://127.0.0.1:43110/1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D/
?

the same error ?

**Comment by HelloZeroNet** [comment]
Please also check if it's the same error in another browser

**Comment by jamesalexanderdickerson** [comment]
Yes, both URLs get the same error. Also the same error in both Firefox and Chrome.

**Comment by jamesalexanderdickerson** [comment]
Yes, both URLs get the same error. Also the same error in both Firefox and Chrome.

**Comment by nomon2009** [comment]
Hello, my first post.

I have encountered the same problem, getting the server error.
The solution that worked for me was going though the "hosts" file in the /etc directory. Creating a copy, deleting the content (in my case it was: 127.0.0.1 lmlicenses.wip4.adobe.com, 127.0.0.1 lm.licenses.adobe.com) than replacing the copy with the original. After entering password for authentication and restarting the computer I was able to connect to ZeroNet yet again. 

Hope this will help you out.

**Comment by HelloZeroNet** [comment]
@jamesalexanderdickerson Do you have any special character in the directory path where you have extracted ZeroNet?

**Comment by jamesalexanderdickerson** [comment]
No special characters. It was first extracted to my Downloads folder. I also tried moving it to my Applications folder. The very first time I ran the application, it appeared to work, and then every time after, I get the error, no matter where it's located.

**Comment by HelloZeroNet** [comment]
Please try to update to latest version, start it, then check `log/debug.log` and search for text "not in allowed dir" it should give more idea about the error.

**Comment by jamesalexanderdickerson** [comment]
Here is the contents of log/debug.log
[2016-03-12 18:33:38,171] DEBUG    PluginManager Loading plugin: AnnounceZero
[2016-03-12 18:33:38,179] DEBUG    PluginManager New plugin registered to: Site
[2016-03-12 18:33:38,179] DEBUG    PluginManager Loading plugin: CryptMessage
[2016-03-12 18:33:38,245] DEBUG    - opensslVerify loaded: <CDLL '/usr/lib/libssl.dylib', handle 10185fa30 at 1039595d0>
[2016-03-12 18:33:38,248] INFO     - OpenSSL loaded, version: 0009081DF
[2016-03-12 18:33:38,249] DEBUG    PluginManager New plugin registered to: UiWebsocket
[2016-03-12 18:33:38,249] DEBUG    PluginManager New plugin registered to: User
[2016-03-12 18:33:38,249] DEBUG    PluginManager Loading plugin: Newsfeed
[2016-03-12 18:33:38,250] DEBUG    PluginManager New plugin registered to: UiWebsocket
[2016-03-12 18:33:38,250] DEBUG    PluginManager New plugin registered to: User
[2016-03-12 18:33:38,251] DEBUG    PluginManager Loading plugin: Sidebar
[2016-03-12 18:33:38,258] DEBUG    PluginManager New plugin registered to: UiRequest
[2016-03-12 18:33:38,258] DEBUG    PluginManager New plugin registered to: UiWebsocket
[2016-03-12 18:33:38,258] DEBUG    PluginManager Loading plugin: Stats
[2016-03-12 18:33:38,266] DEBUG    PluginManager New plugin registered to: UiRequest
[2016-03-12 18:33:38,266] DEBUG    PluginManager Loading plugin: Trayicon
[2016-03-12 18:33:38,266] DEBUG    PluginManager Loading plugin: Zeroname
[2016-03-12 18:33:38,268] DEBUG    PluginManager New plugin registered to: UiRequest
[2016-03-12 18:33:38,269] DEBUG    PluginManager New plugin registered to: SiteManager
[2016-03-12 18:33:38,269] DEBUG    - Config: Config(action='main', batch=False, coffeescript_compiler=None, config_file='zeronet.conf', connected_limit=15, data_dir='data', debug=False, debug_gevent=False, debug_socket=False, disable_encryption=False, disable_sslcompression=True, disable_udp=False, fileserver_ip='*', fileserver_port=15441, homepage='1HeLLo4uzjaLetFx6NH3PMwFP3qbRbTf3D', ip_external=None, keep_ssl_cert=False, log_dir='log', max_files_opened=2048, msgpack_purepython=True, open_browser='default_browser', proxy=None, size_limit=10, stream_downloads=False, tor='enable', tor_controller='127.0.0.1:9051', tor_proxy='127.0.0.1:9050', trackers=['zero://boot3rdez4rzn36x.onion:15441', 'zero://boot.zeronet.io#f36ca555bee6ba216b14d10f38c16f7769ff064e0e37d887603548cc2e64191d:15441', 'udp://tracker.coppersurfer.tk:6969', 'udp://tracker.leechers-paradise.org:6969', 'udp://9.rarbg.com:2710', 'http://tracker.aletorrenty.pl:2710/announce', 'http://explodie.org:6969/announce', 'http://torrent.gresille.org/announce'], trackers_file=False, ui_ip='127.0.0.1', ui_port=43110, ui_restrict=False, use_openssl=True, use_tempfiles=False, verbose=False)
[2016-03-12 18:33:38,269] INFO     - Version: 0.3.6 r1015, Python 2.7.11 |Anaconda 2.5.0 (x86_64)| (default, Dec  6 2015, 18:57:58) 
[GCC 4.2.1 (Apple Inc. build 5577)], Gevent: 1.0.2
[2016-03-12 18:33:38,324] DEBUG    PluginManager New class accepts plugins: SiteManager (Loaded plugins: [<class 'Zeroname.SiteManagerPlugin.SiteManagerPlugin'>, <class 'Site.SiteManager.SiteManager'>])
[2016-03-12 18:33:38,324] DEBUG    PluginManager New class accepts plugins: Site (Loaded plugins: [<class 'AnnounceZero.AnnounceZeroPlugin.SitePlugin'>, <class 'Site.Site.Site'>])
[2016-03-12 18:33:38,334] DEBUG    - Disabled SSL compression on <CDLL '/usr/lib/libssl.dylib', handle 10185fa30 at 103f396d0>
[2016-03-12 18:33:38,347] DEBUG    - Missing SSLwrap, readded.
[2016-03-12 18:33:38,347] DEBUG    - Python SSL version: OpenSSL 1.0.2f  28 Jan 2016
[2016-03-12 18:33:38,379] DEBUG    PluginManager New class accepts plugins: User (Loaded plugins: [<class 'Newsfeed.NewsfeedPlugin.UserPlugin'>, <class 'CryptMessage.CryptMessagePlugin.UserPlugin'>, <class 'User.User.User'>])
[2016-03-12 18:33:38,385] DEBUG    PluginManager New class accepts plugins: UiWebsocket (Loaded plugins: [<class 'Sidebar.SidebarPlugin.UiWebsocketPlugin'>, <class 'Newsfeed.NewsfeedPlugin.UiWebsocketPlugin'>, <class 'CryptMessage.CryptMessagePlugin.UiWebsocketPlugin'>, <class 'Ui.UiWebsocket.UiWebsocket'>])
[2016-03-12 18:33:38,385] DEBUG    PluginManager New class accepts plugins: UiRequest (Loaded plugins: [<class 'Zeroname.UiRequestPlugin.UiRequestPlugin'>, <class 'Stats.StatsPlugin.UiRequestPlugin'>, <class 'Sidebar.SidebarPlugin.UiRequestPlugin'>, <class 'Ui.UiRequest.UiRequest'>])
[2016-03-12 18:33:38,385] INFO     - Creating FileServer....
[2016-03-12 18:33:38,386] DEBUG    TorManager Connecting to 127.0.0.1:9051
[2016-03-12 18:33:38,388] ERROR    TorManager Tor controller connect error: error: [Errno 61] Connection refused in TorManager.py line 151 > socket.py line 344
[2016-03-12 18:33:38,388] DEBUG    TorManager Tor proxy port 127.0.0.1:9050 check error: No connection
[2016-03-12 18:33:38,389] INFO     - Creating UiServer....
[2016-03-12 18:33:38,389] DEBUG    - Loading sites...
[2016-03-12 18:33:38,389] DEBUG    - Added new site: 1Name2NXVi1RDPDgf5617UoW7xA6YrhM9F
[2016-03-12 18:33:38,392] ERROR    Site:1Name2..hM9F Content.json not exist: data/1Name2NXVi1RDPDgf5617UoW7xA6YrhM9F/content.json
[2016-03-12 18:33:38,392] DEBUG    Site:1Name2..hM9F New auth key: b8b6fbdc2b2ec0b38adbf50210917241ee2b945028af4180b2653a319b4ac4a7
[2016-03-12 18:33:38,393] DEBUG    Site:1Name2..hM9F New wrapper key: dddf28d57c504b8b53f53dee01503ce2a860455ae013ff004dfb0fe75142e687
[2016-03-12 18:33:38,394] INFO     - Removing old SSL certs...
[2016-03-12 18:33:38,394] INFO     - Starting servers....
[2016-03-12 18:33:38,394] DEBUG    Site:1Name2..hM9F Start downloading, bad_files: {}, check_size: False, blind_includes: True
[2016-03-12 18:33:38,394] DEBUG    Site:1Name2..hM9F Downloading content.json...
[2016-03-12 18:33:38,394] DEBUG    Site:1Name2..hM9F Need content.json first
[2016-03-12 18:33:38,394] DEBUG    WorkerManager:1Name2..hM9F New task: content.json, peer lock: None, priority: 0, optional_hash_id: None, tasks: 1
[2016-03-12 18:33:38,394] INFO     Ui.UiServer --------------------------------------
[2016-03-12 18:33:38,394] INFO     Ui.UiServer Web interface: http://127.0.0.1:43110/
[2016-03-12 18:33:38,394] INFO     Ui.UiServer --------------------------------------
[2016-03-12 18:33:38,394] INFO     - Opening browser: default_browser...
[2016-03-12 18:33:38,538] DEBUG    - Current RLIMIT_NOFILE: 256, changing to 2048...
[2016-03-12 18:33:38,538] ERROR    Ui.UiServer Web interface bind error, must be running already, exiting.... [Errno 48] Address already in use: ('127.0.0.1', 43110)
[2016-03-12 18:33:38,538] DEBUG    Ui.UiServer Stopped.
[2016-03-12 18:33:39,114] DEBUG    - Generating RSA cert and key PEM files...Generating a 2048 bit RSA private key
.......................................................................................................................................+++
.......................................................................................+++
unable to write 'random state'
writing new private key to 'data/key-rsa.pem'
-----
[2016-03-12 18:33:39,114] DEBUG    FileServer Binding to: *:15441, (msgpack: 0.4.6), supported crypt: ['tls-rsa']
[2016-03-12 18:33:39,114] INFO     FileServer StreamServer bind error, must be running already: [Errno 48] Address already in use: ('', 15441)
[2016-03-12 18:33:39,114] DEBUG    FileServer Stopped.
[2016-03-12 18:33:39,115] DEBUG    Site:1Name2..hM9F Need content.json first

After taking a look at the log, I figured that something else must be running at the port. I used lsof to show anything that was already running on the port. I think I must have been trying to run two instances of ZeroNet at the same time. Thanks for the help.
